bindings/rust/s2n-tls/src/raw/config.rs

```diff
@@ -44,46 +43,49 @@ impl Config {
     /// # Safety
     ///
     /// This config _MUST_ have been initialized with a [`Builder`].
+    /// Additionally, this does NOT increment the config reference count,
+    /// so consider cloning the result if the source pointer is still
+    /// valid and useable afterwards.
     pub(crate) unsafe fn from_raw(config: NonNull<s2n_config>) -> Self {
-        Self(config)
+        let config = Self(config);
+
+        // Check if the context can be retrieved.
+        // If it can't, this is not a valid config.
+        config.context();
```

Comment:
Yes, it will panic. Previously it wouldn't panic here, but any later call that needed the context would panic. This is an unsafe method and the safety comment explains how to avoid the panic.